src/explore-education-statistics-admin/src/pages/release/datablocks/components/chart/__tests__/ChartBuilder.test.tsx

```diff
@@ -0,0 +1,321 @@
+      });
+
+      const tableRows = screen.getAllByRole('row');
+      expect(tableRows).toHaveLength(2);
+      expect(tableRows[1]).toHaveTextContent(
+        'Number of unauthorised absence sessions (Male, All locations, All time periods)',
+      );
+
+      userEvent.selectOptions(
+        screen.getByLabelText('Characteristic'),
+        'female',
+      );
+      userEvent.selectOptions(
+        screen.getByLabelText('Indicator'),
+        'unauthorised-absence-sessions',
+      );
+
+      userEvent.click(screen.getByRole('button', { name: 'Add data set' }));
+
+      await waitFor(() => {
+        expect(screen.getByText('Reorder data sets')).toBeInTheDocument();
+      });
+
+      const updatedTableRows = screen.getAllByRole('row');
+      expect(updatedTableRows).toHaveLength(3);
+      expect(updatedTableRows[2]).toHaveTextContent(
+        'Number of unauthorised absence sessions (Female, All locations, All time periods)',
+      );
+    });
+
+    test('removing data sets', async () => {
+      render(
+        <ChartBuilderFormsContextProvider initialForms={testFormState}>
+          <ChartBuilder
+            releaseId="release-1"
+            data={testTable.results}
+            meta={testSubjectMeta}
+            tableTitle="Table title"
+            onChartSave={jest.fn(() => Promise.resolve())}
+            onChartDelete={noop}
+            onTableQueryUpdate={jest.fn(() => Promise.resolve())}
+          />
+        </ChartBuilderFormsContextProvider>,
+      );
+
+      userEvent.click(screen.getByRole('button', { name: 'Line' }));
+
+      userEvent.click(screen.getByRole('tab', { name: 'Data sets' }));
+
+      expect(
+        screen.getByRole('heading', { name: 'Data sets' }),
+      ).toBeInTheDocument();
+
+      userEvent.click(screen.getByRole('button', { name: 'Add data set' }));
+
+      await waitFor(() => {
+        expect(screen.getByText('Remove all')).toBeInTheDocument();
+      });
+
+      const tableRows = screen.getAllByRole('row');
+      expect(tableRows).toHaveLength(5);
+      expect(tableRows[1]).toHaveTextContent(
+        'Number of authorised absence sessions (Male, All locations, All time periods)',
+      );
+      expect(tableRows[2]).toHaveTextContent(
+        'Number of unauthorised absence sessions (Male, All locations, All time periods)',
+      );
+      expect(tableRows[3]).toHaveTextContent(
+        'Number of authorised absence sessions (Female, All locations, All time periods)',
+      );
+      expect(tableRows[4]).toHaveTextContent(
+        'Number of unauthorised absence sessions (Female, All locations, All time periods)',
+      );
+
+      userEvent.click(
+        within(tableRows[2]).getByRole('button', { name: 'Remove' }),
+      );
+
+      await waitFor(() => {
+        expect(
+          screen.queryByText(
+            'Number of unauthorised absence sessions (Male, All locations, All time periods)',
+          ),
+        ).not.toBeInTheDocument();
+      });
+
```

Comment:
We should check that there are still 3 data sets remaining to ensure there are no bugs that might remove more than one.